libs/messaging/kafka-message-bus-impl/src/main/kotlin/net/corda/messagebus/kafka/serialization/CordaAvroDeserializerImpl.kt

```
@@ -1,60 +1,87 @@
 package net.corda.messagebus.kafka.serialization
 
+import java.nio.ByteBuffer
+import java.util.function.Consumer
 import net.corda.data.CordaAvroDeserializer
+import net.corda.data.chunking.Chunk
+import net.corda.data.chunking.ChunkKey
 import net.corda.schema.registry.AvroSchemaRegistry
+import net.corda.v5.base.exceptions.CordaRuntimeException
+import net.corda.v5.base.util.uncheckedCast
 import org.apache.kafka.common.serialization.Deserializer
 import org.apache.kafka.common.serialization.StringDeserializer
 import org.slf4j.LoggerFactory
-import java.nio.ByteBuffer
-import java.util.function.Consumer
 
 class CordaAvroDeserializerImpl<T : Any>(
     private val schemaRegistry: AvroSchemaRegistry,
     private val onError: Consumer<ByteArray>,
     private val expectedClass: Class<T>
-) : CordaAvroDeserializer<T>, Deserializer<T> {
+) : CordaAvroDeserializer<T>, Deserializer<Any> {
 
     private companion object {
         val stringDeserializer = StringDeserializer()
         val log = LoggerFactory.getLogger(this::class.java.enclosingClass)
+        const val errorMsg = "Failed to deserialize bytes returned"
     }
 
-    override fun deserialize(data: ByteArray): T? {
+    private fun deserialize(data: ByteArray, allowChunks: Boolean = false): Any? {
+        val isChunkType  = allowChunks && isChunkType(data)
```

`isChunkType(data)` will try `schemaRegistry.getClassType(ByteBuffer.wrap(bytes))` which might fail if expectedClass is ByteArray (i.e. it as not Avro serialized)

---

it catches and swallows the exception.